DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/DelegateCoursesController.cs

```diff
@@ -78,5 +83,32 @@ public IActionResult AllCourseStatistics()
 
             return View(model);
         }
+
+        [Route("DownloadAll")]
+        public IActionResult DownloadAll(
+            string? searchString = null,
+            string? sortBy = null,
+            string sortDirection = GenericSortingHelper.Ascending,
+            string? existingFilterString = null
+        )
+        {
+            var centreId = User.GetCentreId();
+            var categoryId = User.GetAdminCourseCategoryFilter();
+            var content = courseDelegatesDownloadFileService.GetCourseDelegateDownloadFile(
+                centreId,
+                categoryId,
+                searchString,
+                sortBy,
+                existingFilterString,
+                sortDirection
+            );
+
+            const string fileName = "Digital Learning Solutions Course Delegates.xlsx";

```

Comment:
Unless this filename has been specified I think it wants some kind of identifying detail, for example the datetime. Right now there'll be no way to distinguish different downloads.